flat/field.go

```diff
@@ -39,9 +39,12 @@ func (f *field) String() string {
 	return f.tag.Get("default")
 }
 
-func (f *field) Get() interface{} {
-	// return f.ptr
-	return nil
+func (f *field) IsZeroValue() bool {
+	fieldType := reflect.TypeOf(f.field.Interface())
+
+	return fieldType == nil ||
+		fieldType.Comparable() &&
+		f.field.Interface() == reflect.Zero(fieldType).Interface()
```

Comment:
```suggestion
func (f *field) IsZero() bool {
	return f.field.IsValid() && f.field.IsZero()
```

The more I think about this, the more I find this harder to make good-enough, because of the way value types are hard to tell if they have been set or it is their _zero_ value.

But at least this way we will be _mirroring_ reflect? The tests passes just fine with this new code.

Let me know if this is sufficient for your use case, then we can merge it. Thanks again!

Reply:
This sounds good!  The only way to tell if a value hasn't been set is to use a pointer - it makes usage clunky and so I'd be happy to skip that consideration.